src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp

```diff
@@ -108,29 +129,229 @@ class jit_convert_array : public jit_kernel {
 
     typedef void (*convert_vec_t)(jit_generator&, const RegExp&, const RegExp&);
 
-    jit_convert_array(convert_vec_t convert_vec, size_t src_size, size_t dst_size)
+    jit_convert_array(convert_vec_t convert_vec, size_t src_size, size_t dst_size, f8_type type, bool is_dst_bf16)
         : jit_kernel(jit_name()),
           _convert_vec(convert_vec),
           _src_size(src_size),
-          _dst_size(dst_size) {}
+          _dst_size(dst_size) {
+        if (type == f8_type::f8e4m3) {
+            f8_e4m3_emu_ = std::make_shared<fp8_emulation_e4m3_t>(this,
+                                                                  fp8_emu_reserv_1_,
+                                                                  fp8_emu_reserv_2_,
+                                                                  fp8_emu_reserv_3_,
+                                                                  fp8_emu_reserv_4_,
+                                                                  fp8_emu_reserv_5_,
+                                                                  fp8_emu_scratch_);
+        } else if (type == f8_type::f8e5m2) {
+            f8_e5m2_emu_ = std::make_shared<fp8_emulation_e5m2_t>(this,
+                                                                  fp8_emu_reserv_1_,
+                                                                  fp8_emu_reserv_2_,
+                                                                  fp8_emu_reserv_3_,
+                                                                  fp8_emu_kmask_aux_,
+                                                                  fp8_emu_scratch_);
+        }
+        if (is_dst_bf16 && mayiuse(cpu_isa_t::avx512_core)) {
+            uni_vcvtneps2bf16_ = std::make_shared<jit_uni_vcvtneps2bf16>(this, cpu_isa_t::avx512_core);
+        }
+    }
 
     template <typename src_t, typename dst_t>
     static fn_t get() {
         if (mayiuse(cpu_isa_t::avx2) && dnnl::impl::cpu::x64::cpu().has(Xbyak::util::Cpu::tF16C)) {
-            static jit_convert_array converter(convert_vec<src_t, dst_t>, sizeof(src_t), sizeof(dst_t));
+            static jit_convert_array converter(convert_vec<src_t, dst_t>,
```

Comment:
I am wondering if we can make jit_convert_array a template parametrized by src_t and dst_t to avoid these new ctor arguments (just to keep future scalability).

Reply:
Good point! Applied. Thanks Dmitry!